src/main/java/org/osc/sdk/controller/api/SdnRedirectionApi.java

```diff
@@ -33,211 +33,230 @@
+     * Creates an inspection hook
      *
-     * @param order
-     *            The order in which to insert the inspection hook. If other inspection hooks are already in place, all
-     *            existing hooks will be push down from the point of insertion. The order range is zero based where -1
-     *            symbolizes to append as last
+     * @param networkElem  provides context information of the network element, see {@link NetworkElement}
+     * @param inspectionPort  provides context information of the inspection port element, see{@link InspectionPortElement}
+     * @param tag  provides a tag
+     * @param encType  provides the tag encapsulation type, see{@link TagEncapsulationType}
+     * @param order  provides an order
+     * @param failurePolicyType  provides the failure policy type, see{@link FailurePolicyType}
+     * @return the identifier of the inspection hook
+     * @throws NetworkPortNotFoundException when port is not found
+     * @throws Exception upon failure
      */
     String installInspectionHook(NetworkElement networkElem, InspectionPortElement inspectionPort, Long tag,
             TagEncapsulationType encType, Long order, FailurePolicyType failurePolicyType)
                     throws NetworkPortNotFoundException, Exception;
 
     /**
-     * Remove previously installed network hooks placed on behalf of network port.
+     * Deletes an inspection hook
      *
-     * @param networkElem
-     *            The port on which behalf traffic is being inspected or a networkElement containing protected ports
-     * @param inspectionPort
-     *            The inspection port belonging to network inspection device.
+     * @param networkElem  provides context information of the network element, see {@link NetworkElement}
+     * @param inspectionPort  provides context information of the inspection port element, see{@link InspectionPortElement}
+     * @throws Exception upon failure
      */
     void removeInspectionHook(NetworkElement networkElem, InspectionPortElement inspectionPort) throws Exception;
 
     /**
-     * Remove the previously installed inspection hook given its identifier.
+     * Deleted an inspection hook
      *
-     * @param inspectionHookId
-     *            The identifier of the inspection hook to be removed.
+     * @param inspectionHookId  the identifier of the inspection hook
+     * @throws Exception upon failure
      */
     void removeInspectionHook(String inspectionHookId) throws Exception;
 
     /**
-     * Retrieves a previously installed inspection hook given its identifier.
+     * Retrieves an InspectionHookElement
      *
-     * @param inspectionHookId
-     *            The identifier of the inspection hook to be retrieved.
+     * @param inspectionHookId  the identifier of the inspection hook
+     * @return the inspection hook element, see {@link InspectionHookElement}
```

Comment:
If there is none found should it return null? Take a look a the OSC core code and see what is the expectation. The javadoc should indicate that null is the return value if nothing is found (if that is what OSC expects)